torchgeo/datasets/ssl4eo_benchmark.py

```diff
@@ -0,0 +1,369 @@
+            mask_product: mask target one of ['cdl', 'nlcd']
+            split: dataset split, one of ['train', 'val', 'test']
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 after downloading files (may be slow)
+
+        Raises:
+            AssertionError: if any arguments are invalid
+        """
+        assert (
+            input_sensor in self.valid_input_sensors
+        ), f"Only supports one of {self.valid_input_sensors}, but found {input_sensor}."
+        self.input_sensor = input_sensor
+        assert (
+            mask_product in self.valid_mask_products
+        ), f"Only supports one of {self.valid_mask_products}, but found {mask_product}."
+        self.mask_product = mask_product
+        assert (
+            split in self.valid_splits
+        ), f"Only supports one of {self.valid_splits}, but found {split}."
+        self.split = split
+
+        self.root = root
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+        self.img_dir_name = self.image_root.format(self.input_sensor)
+        self.mask_dir_name = self.mask_dir_dict[self.input_sensor].format(
+            self.mask_product
+        )
+
+        self._verify()
+
+        self.sample_collection = self.retrieve_sample_collection()
+
+        # train, val, test split
+        np.random.seed(0)
+        sizes = (np.array(self.split_percentages) * len(self.sample_collection)).astype(
+            int
+        )
+        cutoffs = np.cumsum(sizes)[:-1]
+        sample_indices = np.arange(len(self.sample_collection))
+        np.random.shuffle(sample_indices)
+        groups = np.split(sample_indices, cutoffs)
+        split_indices = {"train": groups[0], "val": groups[1], "test": groups[2]}[
+            self.split
+        ]
+
+        self.sample_collection = [self.sample_collection[idx] for idx in split_indices]
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        img_pathname = os.path.join(self.root, self.img_dir_name, "**", "all_bands.tif")
+        exists = []
+        for fname in glob.iglob(img_pathname, recursive=True):
+            exists.append(True)
+            break
+        mask_pathname = os.path.join(
+            self.root,
+            self.mask_dir_name,
+            "**",
+            f"{self.mask_product}_{self.year_dict[self.input_sensor]}.tif",
+        )
+        for fname in glob.iglob(mask_pathname, recursive=True):
+            exists.append(True)
+            break
+        if exists and all(exists):
+            return
```

Comment:
Won't this be true if image is found or mask is found but also if only one or the other is found?